### PR TITLE
[engraving] Renamed layout subsystem to rendering. Step 34

### PR DESCRIPTION
**Pull request by igorkorsukov**
And moved draw Accidental draw

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
I think we should not mix layout and draw too much. Of course, since they are so inseparable, it makes sense to put them nearby each other; but they are still two separate steps of the process, and should not get mixed. Therefore I'm inclined to suggest to put layout and draw in two separate subfolders in a rendering subsystem, and _not_ to put them in the same `IRendering` class.

**Comment by igorkorsukov** [comment]
> I think we should not mix layout and draw too much. Of course, since they are so inseparable, it makes sense to put them nearby each other; but they are still two separate steps of the process, and should not get mixed. Therefore I'm inclined to suggest to put layout and draw in two separate subfolders in a rendering subsystem, and _not_ to put them in the same `IRendering` class.

On the contrary, I am more inclined to combine them closely, for example, for each type of element, make a class of something like `SomeNameRendering` (ex. AccidentalRendering) and place both layout and draw there. This will allow us to simply see what is being done in the layout and, accordingly, what the draw can use.

The second point, as we talk, layout for palettes and drag is essentially the same thing, it makes sense for them to do something like `rendering/single`, which can be used somewhere else where we just need to draw one element. So, in this case, we may not show the layout stage in the public interface, provide only a draw, which, if necessary, will do a layout.
i.e. there will be something like
```
ISingleRendering::drawItem(EngravingItem* item, ...)
{
        if (!item->layoutData().isValid()) {
                layout(item);
        }

        doDrawItem(item, ...);
}
```

The third point, we need to somehow clearly show that layout and draw are inseparable, we cannot call draw without first calling layout - I think this is the reason for keeping them in the same interface.

**Comment by igorkorsukov** [comment]
Moved palettelayout to engraving/rendering/single

File: src/palette/internal/palettedraw.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        item->draw(painter);
+    }
+}
+
+void PaletteDraw::draw(const Accidental* item, draw::Painter* painter)
+{
+    // don't show accidentals for tab or slash notation
+    if (item->onTabStaff() || (item->note() && item->note()->fixed())) {
+        return;
+    }
```
Comment (cbjeukendrup) [inline src/palette/internal/palettedraw.cpp]: (This is not strictly necessary in the Palettes, of course; but perhaps we should just focus on moving code for now, and do refactoring later.)

File: src/engraving/rendering/dev/layout.h [hunk 2]
```diff
@@ -28,8 +28,8 @@ namespace mu::engraving {
 class Score;
 }
 
-namespace mu::engraving::layout::dev  {
-class Layout : public ILayout
+namespace mu::engraving::rendering::dev  {
+class Layout : public IRendering
```
Comment (cbjeukendrup) [inline src/engraving/rendering/dev/layout.h]: Why are we not renaming this class too?

File: src/engraving/rendering/stable/layout.h [hunk 3]
```diff
@@ -28,8 +28,8 @@ namespace mu::engraving {
 class Score;
 }
 
-namespace mu::engraving::layout::stable  {
-class Layout : public ILayout
+namespace mu::engraving::rendering::stable  {
+class Layout : public IRendering
```
Comment (cbjeukendrup) [inline src/engraving/rendering/stable/layout.h]: (Same here)

**Review by cbjeukendrup — APPROVED** [review]
Ignore my previous "review" (if you did receive it); it was outdated, but there seems to be a bug in GitHub that when you click "Cancel review" it actually publishes the pending comments instead of cancelling.

**Pull request merged**